internal/testing/pruning_test.go

```diff
@@ -190,12 +199,55 @@ func TestSectorPruning(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	// create a contracts dict
+	c, err := b.Contracts(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	contracts := make(map[types.PublicKey]api.ContractMetadata)
+	for _, contract := range c {
+		contracts[contract.HostKey] = contract
+	}
+
 	// add an object
 	if err := w.UploadObject(context.Background(), bytes.NewReader([]byte(t.Name())), "obj"); err != nil {
 		t.Fatal(err)
 	}
 
+	// fetch the object and for every shard fetch the contract sectors and compare them
+	obj, _, err := b.Object(context.Background(), "obj")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(obj.Slabs) == 0 {
+		t.Fatal("expected at least one slab")
+	}
+	for _, shard := range obj.Slabs[0].Shards {
+		contract, ok := contracts[shard.Host]
+		if !ok {
+			t.Fatal("could not find contract for host")
+		}
+		roots, err := w.RHPContractRoots(context.Background(), contract.ID)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !hasRoot(roots, shard.Root) {
+			t.Fatal("root not found in contract", len(roots), shard.Host, shard.Root)
+		}
+
+		roots, err = b.ContractRoots(context.Background(), contract.ID)
```

Comment:
Aren't you fetching the same roots twice here?

Reply:
`worker.RHPContractRoots`: returns contract roots straight from the `host`
`bus.ContractRoots`: returns the contract's roots from the database
it's just a simple assertion of both methods in an integration test